app/src/main/java/com/futurice/freesound/feature/audio/BaseAudioPlayerEventListener.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 Futurice GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.futurice.freesound.feature.audio;
+
+import com.google.android.exoplayer2.ExoPlaybackException;
+import com.google.android.exoplayer2.ExoPlayer;
+import com.google.android.exoplayer2.Timeline;
+import com.google.android.exoplayer2.source.TrackGroupArray;
+import com.google.android.exoplayer2.trackselection.TrackSelectionArray;
+
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.reactivex.Observer;
+import io.reactivex.disposables.Disposable;
+
+import static com.futurice.freesound.utils.Preconditions.get;
+
+/**
+ * Base class for making Observables from ExoPlayer callback events.
+ *
+ * @param <T> the Observable value type.
+ */
+abstract class BaseAudioPlayerEventListener<T> implements Disposable, ExoPlayer.EventListener {
```

Comment:
I don't want instances of this class to be created, it should always be subclassed (there will be more subclasses in the future).